### PR TITLE
[REVIEW] Update gcc to 7.3 and numba to 0.44

### PR DESCRIPTION
**Pull request by dillon-cullinan**
Updating GCC to 7.3. Tested cudf source build and no failures.

### CODE REVIEW

**Comment by dillon-cullinan** [comment]
Tested new numba and gcc version on branch-0.8 of cuDF and everything built okay.

File: .condarc-cuda10.0 [hunk 1]
```diff
@@ -2,5 +2,6 @@ channels:
   - rapidsai/label/cuda10.0
   - rapidsai-nightly/label/cuda10.0
   - nvidia/label/cuda10.0
+  - numba
```
Comment (kkraus14) [inline .condarc-cuda10.0]: Numba 0.44 is released on conda-forge and defaults now so this is no longer needed
```suggestion
```

File: .condarc-cuda9.2 [hunk 2]
```diff
@@ -2,5 +2,6 @@ channels:
   - rapidsai/label/cuda9.2
   - rapidsai-nightly/label/cuda9.2
   - nvidia/label/cuda9.2
+  - numba
```
Comment (kkraus14) [inline .condarc-cuda9.2]:
```suggestion
```

File: Dockerfile.centos7 [hunk 3]
```diff
@@ -106,14 +106,14 @@ RUN rm -rf ${RAPIDS_SRC_DIR}/tmp
 ARG GCC7_DIR=${RAPIDS_SRC_DIR}/gcc7
 
 RUN mkdir -p ${GCC7_DIR}
-RUN cd ${GCC7_DIR} && wget -q http://ftp.gnu.org/gnu/gcc/gcc-7.2.0/gcc-7.2.0.tar.gz 
-RUN cd ${GCC7_DIR} && tar zxf gcc-7.2.0.tar.gz
-RUN cd ${GCC7_DIR}/gcc-7.2.0 && \
+RUN cd ${GCC7_DIR} && wget -q http://ftp.gnu.org/gnu/gcc/gcc-7.2.0/gcc-7.3.0.tar.gz
```
Comment (kkraus14) [inline Dockerfile.centos7]: This path is mixing 7.2 and 7.3 oddly

**Comment by mike-wendt** [comment]
- [x] gcc7.3 is in the new containers already
- [ ] numba 0.44 is not

@kkraus14 and @dillon-cullinan if we need 0.44 numba in the new containers I can add it and that would complete this PR

**Comment by mike-wendt** [comment]
Put numba 0.44 into new containers

**Pull request merged**